### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by Piedone** [ubicloud/ubicloud#1172] [Q&A] [answered]
Does the Build Your Own Cloud approach support Windows?

Thanks!

**Comment by byucesoy** [comment, marked as answer]
Hi @Piedone,

Most complex step described on Build your own cloud section is cloudifying the given servers, which runs on the docker container, so that should work without any issue given that there is a docker environment. Only step that runs before bringing containers up is generating some environment variables. [This](https://github.com/ubicloud/ubicloud/blob/6e56364d17d503d6a31c18a19372dbf100a6a4ab/demo/generate_env) is the script that generates those environment variables. It is quite simple and can be replicated for Windows environment easily or even you can manually set those environment variables.

However, it should be noted that these docker containers are intended for demo purposes or getting familiar with the system. I'd recommend running everything on native Linux environment for having more performant and reliable system.

**Reply by Piedone** [reply]
I see, thank you.
